beacon-chain/blockchain/forkchoice/process_attestation.go

```diff
@@ -151,17 +151,19 @@ func (s *Store) saveCheckpointState(ctx context.Context, baseState *pb.BeaconSta
 	// Advance slots only when it's higher than current state slot.
 	if helpers.StartSlot(c.Epoch) > baseState.Slot {
 		stateCopy := proto.Clone(baseState).(*pb.BeaconState)
-		baseState, err = state.ProcessSlots(ctx, stateCopy, helpers.StartSlot(c.Epoch))
+		stateCopy, err = state.ProcessSlots(ctx, stateCopy, helpers.StartSlot(c.Epoch))
 		if err != nil {
 			return nil, errors.Wrapf(err, "could not process slots up to %d", helpers.StartSlot(c.Epoch))
 		}
-	}
 
-	if err := s.checkpointState.AddCheckpointState(&cache.CheckpointState{
-		Checkpoint: c,
-		State:      baseState,
-	}); err != nil {
-		return nil, errors.Wrap(err, "could not saved checkpoint state to cache")
+		if err := s.checkpointState.AddCheckpointState(&cache.CheckpointState{
```

Comment:
Please add unit test that checkpoint should only be saved if start slot of the checkpoint is greater than the base state slot. This is a logic test and if it’s a bug then it should have regression test please.

Reply:
Done